README.md

```diff
@@ -23,25 +23,129 @@ import {
 } from 'react-native';
 import InfiniteScrollView from 'react-native-infinite-scroll-view';
 
+class ExampleComponent extends React.Component {
+  async _loadMoreContentAsync() {
+    // Fetch more data here.
+  }
 
-// Inside of a component's render() method:
-render() {
-  return (
-    <ListView
-      renderScrollComponent={props => <InfiniteScrollView {...props} />}
-      dataSource={...}
-      renderRow={...}
-      canLoadMore={this.state.canLoadMoreContent}
-      onLoadMoreAsync={this._loadMoreContentAsync()}
-    />
-  );
+  render() {
+    return (
+      <ListView
+        renderScrollComponent={props => <InfiniteScrollView {...props} />}
+        dataSource={...}
+        renderRow={...}
+        canLoadMore={this.state.canLoadMoreContent}
+        onLoadMoreAsync={this._loadMoreContentAsync.bind(this)}
+      />
+    );
+  }
 }
 ```
 
+A more complete example that uses a `ListView.DataSource`, [react-redux](https://github.com/reactjs/react-redux), and supports pagination would look something like this:
+
+```js
+import React from 'react';
+import {
+  ListView,
+  RefreshControl,
+} from 'react-native';
+import InfiniteScrollView from 'react-native-infinite-scroll-view';
+
+class ExampleComponent extends React.Component {
+  static propTypes = {
+    // Assume data shape looks like:
+    // {items: ["item1", "item2"], nextUrl: null, isFetching: false}
+    listData: PropTypes.object.isRequired,
+  }
+
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      dataSource: new ListView.DataSource({
+        rowHasChanged: this._rowHasChanged.bind(this),
+      }),
+    };
+  }
+
+  async componentWillMount() {
+    // Initial fetch for data, assuming that listData is not yet populated.
+    this._loadMoreContentAsync();
+  }
+
+  componentDidMount() {
+    // Trigger a re-render when receiving initial props.
+    this.updateDataSource(this.props);
```

Comment:
The data source should be updated in the constructor so that the data is rendered on the first render call. A clean way to write this is to define `updateDataSource(dataSource, props)` and call it from the constructor and `componentWillReceiveProps`